packages/platform-server/test/incremental_hydration_spec.ts

```diff
@@ -1514,5 +1520,233 @@ describe('platform-server partial hydration integration', () => {
+
+    it('should clear registry of blocks as they are hydrated', async () => {
+      @Component({
+        standalone: true,
+        selector: 'app',
+        template: `
+          <main (click)="fnA()">
+            @defer (on viewport; hydrate on interaction) {
+              <div id="main" (click)="fnA()">
+                Main defer block rendered!
+                @defer (on viewport; hydrate on interaction) {
+                  <p id="nested">Nested defer block</p>
+                } @placeholder {
+                  <span>Inner block placeholder</span>
+                }
+              </div>
+            } @placeholder {
+              <span>Outer block placeholder</span>
+            }
+          </main>
+        `,
+      })
+      class SimpleComponent {
+        fnA() {}
+
+        registry: DehydratedBlockRegistry = inject(DEHYDRATED_BLOCK_REGISTRY);
+        jsActionMap: Map<string, Set<Element>> = inject(JSACTION_BLOCK_ELEMENT_MAP);
+        contract: EventContractDetails = inject(JSACTION_EVENT_CONTRACT);
+      }
+
+      const appId = 'custom-app-id';
+      const providers = [{provide: APP_ID, useValue: appId}];
+      const hydrationFeatures = () => [withIncrementalHydration()];
+
+      const html = await ssr(SimpleComponent, {envProviders: providers, hydrationFeatures});
+
+      // Internal cleanup before we do server->client transition in this test.
+      resetTViewsFor(SimpleComponent);
+
+      ////////////////////////////////
+      const doc = getDocument();
+
+      const appRef = await prepareEnvironmentAndHydrate(doc, html, SimpleComponent, {
+        envProviders: [...providers, {provide: PLATFORM_ID, useValue: 'browser'}],
+        hydrationFeatures,
+      });
+      const compRef = getComponentRef<SimpleComponent>(appRef);
+      appRef.tick();
+      await whenStable(appRef);
+
+      const registry = compRef.instance.registry;
+      const jsActionMap = compRef.instance.jsActionMap;
+      const contract = compRef.instance.contract;
+      spyOn(contract.instance!, 'cleanUp').and.callThrough();
+
+      expect(registry.size).toBe(1);
+      expect(jsActionMap.size).toBe(2);
+      expect(registry.has('d0')).toBeTruthy();
+
+      const mainBlock = doc.getElementById('main')!;
+      const clickEvent = new CustomEvent('click', {bubbles: true});
+      mainBlock.dispatchEvent(clickEvent);
+
+      await timeout(1000); // wait for defer blocks to resolve
+
+      appRef.tick();
+      expect(registry.size).toBe(1);
+      expect(registry.has('d0')).toBeFalsy();
+      expect(jsActionMap.size).toBe(1);
+
+      const nested = doc.getElementById('nested')!;
+      const clickEvent2 = new CustomEvent('click', {bubbles: true});
+      nested.dispatchEvent(clickEvent2);
+      await timeout(1000); // wait for defer blocks to resolve
+      appRef.tick();
+
+      expect(registry.size).toBe(0);
+      expect(jsActionMap.size).toBe(0);
+      expect(contract.instance!.cleanUp).toHaveBeenCalled();
+    }, 100_000);
```

Comment:
Was that timeout helpful for testing/debugging only and can be removed?
```suggestion
    });
```